util_codes/View.py

```diff
@@ -0,0 +1,1128 @@
+            elif view =="Y" or view=="y":
+                val_a, val_b = np.log10(np.abs(self.VAL_Y)), phase(self.VAL_Y)
+            elif view =="Z" or view=="z":
+                val_a, val_b = np.log10(np.abs(self.VAL_Z)), phase(self.VAL_Z)
+        else:
+            if view =="X" or view=="x":
+                val_a, val_b = np.abs(self.VAL_X), phase(self.VAL_X)
+            elif view =="Y" or view=="y":
+                val_a, val_b = np.abs(self.VAL_Y), phase(self.VAL_Y)
+            elif view =="Z" or view=="z":
+                val_a, val_b = np.abs(self.VAL_Z), phase(self.VAL_Z)
+
+        if self.normal =="X" or self.normal=="x":
+            a, b = self.y, self.z
+            vec_a, vec_b = self.VAL_Y, self.VAL_Z
+            xlabel = "Y (m)"
+            ylabel = "Z (m)"
+        elif self.normal =="Y" or self.normal =="y":
+            a, b = self.x, self.z
+            vec_a, vec_b = self.VAL_X, self.VAL_Z
+            xlabel = "X (m)"
+            ylabel = "Z (m)"
+        elif self.normal =="Z" or self.normal =="z":
+            a, b = self.x, self.y
+            vec_a, vec_b = self.VAL_X, self.VAL_Y
+            xlabel = "X (m)"
+            ylabel = "Y (m)"
+
+        if clim == None:
+            vamin, vamax = val_a.min(), val_a.max()
+            vbmin, vbmax = val_b.min(), val_b.max()
+        else:
+            vamin, vamax = clim[0][0], clim[0][1]
+            vbmin, vbmax = clim[1][0], clim[1][1]
+
+        dat0 = ax0.contourf(a, b, val_a, ncontour, clim=(vamin,vamax), vmin=vamin, vmax=vamax)
+        dat1 = ax1.contourf(a, b, val_b, ncontour, clim=(vbmin,vbmax), vmin=vbmin, vmax=vbmax)
+
+        if showcontour:
+            ncontours = 1
+            levels_a = vamax*cont_perc
+            levels_b = vbmax*cont_perc
+            ax0.contour(a, b, val_a, ncontours, level=levels_a, colors="k")
+            ax1.contour(a, b, val_b, ncontours, level=levels_b, colors="k")
+
+        cb0 = plt.colorbar(dat0,ax=ax0, format="%.1e", ticks=np.linspace(vamin, vamax, 3))
+        cb1 = plt.colorbar(dat1,ax=ax1, format="%.1e", ticks=np.linspace(vbmin, vbmax, 3))
+        ax0.set_xlabel(xlabel)
+        ax0.set_ylabel(ylabel)
+        ax1.set_xlabel(xlabel)
+        # ax1.set_ylabel(ylabel)
+
+        if view == "vec":
+            ax0.streamplot(a, b, np.abs(vec_a),  np.abs(vec_b), color="w", linewidth=0.5)
+            ax1.streamplot(a, b, phase(vec_a),  phase(vec_b), color="w", linewidth=0.5)
+
+        plt.show()
+
+    
+    def plot_1D_RI_f_x(self,absloc,coordloc,ax0,ax1,sigind):
+
+        if self.normal.upper() == "Z":
+            obsLoc=np.c_[absloc,coordloc,self.z]
+        elif self.normal.upper() == "Y":
+            obsLoc=np.c_[absloc,self.y,coordloc]
+        elif self.normal.upper() == "X":
+            obsLoc=np.c_[self.x,absloc,coordloc]
+
+        self.eval_loc(self.srcLoc,obsLoc, self.sigvec, self.fvec, self.orientation,self.func)
+        
+        ax0.set_xlabel("Frequency (Hz)")
```

Comment:
we can perhaps take these as dictionary entries https://github.com/simpeg/simpeg/blob/master/SimPEG/Mesh/View.py#L203